core/src/main/kotlin/org/evomaster/core/output/PartialOracles.kt

```diff
@@ -32,15 +35,29 @@ class PartialOracles {
     private var oracles = mutableListOf(SupportedCodeOracle(), SchemaOracle())
     private val expectationsMasterSwitch = "ems"
 
-    fun variableDeclaration(lines: Lines, format: OutputFormat){
+    /**
+     * The [variableDeclaration] method handles the generation of auxiliary variables for the partial oracles.
+     * The parameters [lines] and [format] are received from the [TestCaseWriter].
+     * The parameter [active] records which partial oracles are active (i.e. which actually generate
+     * a failing expectation). [active] is a mutable map, where the key is a string containing the name of
+     * an [ImplementedOracle], and the key is a boolean indicating whether the oracle is active (i.e. will
+     * generate a failing expectation, and thus require the additional variables) or inactive (i.e. it will
+     * not generate a failing expectation and thus can be skipped).
+     *
+     * The goal of this method is to ensure that only relevant variables are generated (i.e. to avoid
+     * generating stub variables that are never used).
+     */
+    fun variableDeclaration(lines: Lines, format: OutputFormat, active: MutableMap<String, Boolean>){
         for (oracle in oracles){
-            oracle.variableDeclaration(lines, format)
+            if(active.get(oracle.getName()) == true) {
```

Comment:
if on `active` you only call `get`, why did you choose to have it mutable? 
recall that `MutableMap` extends `Map`, so that should be the type as input (as more restrictive)